components/admin/lessons/AdminModuleInputs.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useState } from 'react'
+import {
+  AddModuleMutation,
+  UpdateModuleMutation,
+  useAddModuleMutation,
+  useUpdateModuleMutation
+} from '../../../graphql'
+import { formChange } from '../../../helpers/formChange'
+import { FormCard, MD_INPUT, TextField } from '../../FormCard'
+import { AlertFillIcon, CheckCircleIcon } from '@primer/octicons-react'
+import styles from '../../../scss/adminModuleInputs.module.scss'
+import { Spinner } from 'react-bootstrap'
+import { get } from 'lodash'
+import { RefetchQueriesFunction, ApolloError } from '@apollo/client'
+
+type Module = { id: number; name: string; content: string }
+
+type Props = {
+  lessonId: number
+  refetch: RefetchQueriesFunction
+  title?: string
+  onAddModule?: (
+    m:
+      | (AddModuleMutation['addModule'] & { lesson: { id: number } })
+      | (UpdateModuleMutation['updateModule'] & { lesson: { id: number } })
+      | null,
+    e: { name: string; content: string } | null
+  ) => void
+  module?: Module | undefined
+}
+
+enum Error {
+  InvalidData = 'missing module name or description'
+}
+
+const initValues = (
```

Comment:
This has been turned into a function to auto-fill the forms based on the `module` prop data.